### PR TITLE
Do not MERGE - Script to add backing devices

### PR DESCRIPTION
**Pull request by hathyaga**
This script will add and remove Backing devices
to a Network virtualized device

Signed-off-by: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by hathyaga** [comment]
Job log 
[job-log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1853696/job-log.txt)

**Comment by hathyaga** [comment]
I need a review here with parsing the output to determine pass/fail in network_virtualization_backing_device_add()

**Comment by narasimhan-v** [comment]
@hathyaga Consider adding on the same script with https://github.com/avocado-framework-tests/avocado-misc-tests/blob/master/io/net/virt-net/Network_virtualization_dlpar.py

File: io/net/virt-net/Network_virtualization_Backing_device_add_remove.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+            self.network_virtualization_backing_device_remove()
+
+    def network_virtualization_backing_device_remove(self):
+        '''
+        Network virtualized port remove operation
+        '''
+        self.changehwres(self.server, 'r', self.lpar_id, self.slot_num,
+                         self.backing_devices, 'remove')
+        output = self.listhwres(self.server, self.lpar)
+        # To Do: parse the output for pass/fail
```
Comment (narasimhan-v) [inline io/net/virt-net/Network_virtualization_Backing_device_add_remove.py]: The output needs to be taken before add/remove and after add/remove.
Then, the count of sriov needs to be compared.

Comment (narasimhan-v) [reply io/net/virt-net/Network_virtualization_Backing_device_add_remove.py]: @hathyaga Take the output of listhwres() and find the count. Now do this before and after add/remove.

Comment (narasimhan-v) [reply io/net/virt-net/Network_virtualization_Backing_device_add_remove.py]: @hathyaga If the output is given, then I can say it it looks fine or not. Not with code snippet alone.

Comment (hathyaga) [reply io/net/virt-net/Network_virtualization_Backing_device_add_remove.py]: The job.log has the o/p.  it lists the backing devices before and after backing device add/remove.

Comment (hathyaga) [reply io/net/virt-net/Network_virtualization_Backing_device_add_remove.py]: My concern is what i do to validate that data in function listhwres(self, server, lpar):
Can you check that?

Comment (narasimhan-v) [reply io/net/virt-net/Network_virtualization_Backing_device_add_remove.py]:
```
Running the command on hmc chhwres -r virtualio --rsubtype vnic -o s -m ltc-zz3 -s 4 --id 15                    -a backing_devices+=sriov/ltc-zz3-vios1/100/1/0/4
Running the command on hmc lshwres -r virtualio -m  ltc-zz3 --rsubtype vnic --level lpar --filter                "lpar_names=ltc-zz3-lp15-vnic-automation" -F slot_num,backing_device_states
[stdout] ['3,"sriov/27004037/1/Operational,sriov/27004039/0/Operational"', '4,"sriov/27004038/1/Operational,sriov/27004043/0/Operational"']
[stdout] 

Running the command on hmc chhwres -r virtualio --rsubtype vnic -o s -m ltc-zz3 -s 4 --id 15                    -a backing_devices-=sriov/ltc-zz3-vios1/100/1/0/4
Running the command on hmc lshwres -r virtualio -m  ltc-zz3 --rsubtype vnic --level lpar --filter                "lpar_names=ltc-zz3-lp15-vnic-automation" -F slot_num,backing_device_states
[stdout] ['3,"sriov/27004037/1/Operational,sriov/27004039/0/Operational"', '4,sriov/27004038/1/Operational']
[stdout]
```

Comment (narasimhan-v) [reply io/net/virt-net/Network_virtualization_Backing_device_add_remove.py]: The command looks good, I thought we already discussed and finalised on that.
Count on that needs to be taken, as mentioned.

File: io/net/virt-net/Network_virtualization_Backing_device_add_remove.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+    def listhwres(self, server, lpar):
+        cmd = 'lshwres -r virtualio -m  %s --rsubtype vnic --level lpar --filter \
+               "lpar_names=%s" -F slot_num,backing_device_states' \
+               % (server, lpar)
+        try:
+            output = self.run_command(cmd)
+            print output
+            backing_dev_list = []
+            for i in output:
```
Comment (narasimhan-v) [inline io/net/virt-net/Network_virtualization_Backing_device_add_remove.py]: pylint please

File: io/net/virt-net/Network_virtualization_Backing_device_add_remove.py (hunk 1 above)
Comment (narasimhan-v) [reply io/net/virt-net/Network_virtualization_Backing_device_add_remove.py]: I assume here the `output = ['27004038']
Isn't it ?

**Comment by hathyaga** [comment]
New PR sent: https://github.com/avocado-framework-tests/avocado-misc-tests/pull/842

**Pull request closed without merging**